src/content/docs/2/guide/plugins/develop-mobile.mdx

```diff
@@ -0,0 +1,394 @@
+
+```rust
+// lib.rs
+use tauri::Runtime;
+
+impl<R: Runtime> <plugin-name><R> {
+  pub fn do_something(&self) {
+    // do something that is a shared implementation between desktop and mobile
+  }
+}
+```
+
+This implementation simplifies the process of sharing an API that can be used both by commands and Rust code.
+
+### Develop an Android Plugin
+
+A Tauri plugin for Android is defined as a Kotlin class that extends `app.tauri.plugin.Plugin` and is annoted with `app.tauri.annotation.TauriPlugin`. Each method annotated with `app.tauri.annotation.Command` can be called by Rust or JavaScript.
+
+Tauri uses Kotlin by default for the Android plugin implementation, but you can switch to Java if you prefer. After generating a plugin, right click the Kotlin plugin class in Android Studio and select the "Convert Kotlin file to Java file" option from the menu. Android Studio will guide you through the project migration to Java.
+
+### Develop an iOS Plugin
+
+TODO: 
+
+
+## Plugin Configuration
+
+Refer to the [Plugin Configuration section](/2/guide/plugins/#plugin-configuration) of the Plugin Development guide for more details on developing plugin configurations.
+
+The plugin instance on mobile has a getter for the plugin configuration:
+
+<Tabs>
+<TabItem label="Android">
+
+```java
+@TauriPlugin
+class ExamplePlugin(private val activity: Activity): Plugin(activity) {
+  override fun load(webView: WebView) {
+    val timeout = this.config.getInt("timeout", 30)
+  }
+}
+```
+</TabItem>
+<TabItem label="iOS">
+
+```swift
+class ExamplePlugin: Plugin {
+  @objc public override func load(webview: WKWebView) {
+    let timeout = self.config["timeout"] as? Int ?? 30
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Lifecycle Events
+
+Plugins can hook into several lifecycle events:
+
+- [load](#load): When the plugin is loaded into the web view
+- [onNewIntent](#onnewintent): Android only, when the activity is re-launched
+
+There are also the additional [lifecycle events for plugins](/2/guide/plugins#lifecycle-events) in the Plugin Development guide.
+
+### load
+
+- **When**: When the plugin is loaded into the web view
+- **Why**: TODO:
+
+{/* TODO: above */}
+
+<Tabs>
+<TabItem label="Android">
+
+```java
+@TauriPlugin
+class ExamplePlugin(private val activity: Activity): Plugin(activity) {
+  override fun load(webView: WebView) {
+    // perform plugin setup here
+  }
+}
+```
+
+</TabItem>
+<TabItem label="iOS">
+
+```swift
+class ExamplePlugin: Plugin {
+  @objc public override func load(webview: WKWebView) {
+    let timeout = self.config["timeout"] as? Int ?? 30
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### onNewIntent
+
+**Note**: This is only available on Android.
+
+- **When**: When the activity is re-launched. See [Activity#onNewIntent](<https://developer.android.com/reference/android/app/Activity#onNewIntent(android.content.Intent)>) for more information.
+- **Why**: TODO:
```

Comment:
TODO

Reply:
can you please stop asking me why? :joy:

Reply:
NEVER!!!!! 😆